tapgarden/seedling.go

```diff
@@ -95,6 +97,22 @@ type Seedling struct {
 
 	// update is used to send updates w.r.t the state of the batch.
 	updates SeedlingUpdates
+
+	// ScriptKey is the tweaked Taproot key that will be used to spend the
+	// asset after minting. By default, this key is constructed with a
+	// BIP-0086 style tweak.
+	ScriptKey *asset.ScriptKey
+
+	// GroupInternalKey is the raw group key before the tweak with the
+	// genesis point or tapscript root has been applied.
+	GroupInternalKey *keychain.KeyDescriptor
+
+	// TapscriptRoot is the root of the Tapscript tree that commits to all
+	// script spend conditions for the group key. Instead of spending an
+	// asset, these scripts are used to define witnesses more complex than
+	// a Schnorr signature for reissuing assets. A group key with an empty
+	// Tapscript root can only authorize reissuance with a signature.
+	GroupTapscriptRoot []byte
```

Comment:
Any reason to not make this a `[32]byte`? Then we won't need the checks below.

Reply:
Good point, I don't think so.

Reply:
Thinking about it further, the reason we use `[]byte` for the tapscript root elsewhere is to not store the all-zero byte slice by default / for all cases where there is no root.